src/executor/delete_executor.cpp

```diff
@@ -85,6 +86,14 @@ bool DeleteExecutor::DExecute() {
 
   auto current_txn = executor_context_->GetTransaction();
 
+  oid_t table_oid = target_table_->GetOid();
+  // Lock the table (reader lock)
+  concurrency::LockManager *lm = concurrency::LockManager::GetInstance();
+  bool lock_success = lm->LockShared(table_oid);
+  if (!lock_success) {
+    LOG_WARN("Cannot obtain lock for the table, abort!");
```

Comment:
Make log level consist for same scenario, we use LOG_TRACE in some places.

Reply:
Sorry, we used warn to debug and I missed this when changing everything back before PR. Fixed in local branch already